project4/name_server/tasks/nameserver_task.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * @file Name Server Task
+ */
+#ifndef PROJECT4_NAMESERVER_TASK_H
+#define PROJECT4_NAMESERVER_TASK_H
+
+#include <iostream>
+#include "thread_pool/task.h"
+#include "../nameserver.h"
+#include "../commands.h"
+#include "../../../common/queue/queue.h"
+
+namespace nameserver_tasks {
+
+class NameserverTask : public thread_pool::Task {
```

Comment:
Once again, can we unify the `NameserverTask` and `BootstrapTask` classes? See my previous comments about unifying the driver classes for details.